pkg/controller/machineset/machineset_controller_test.go

```diff
@@ -131,6 +140,71 @@ var _ = Describe("MachineSet Reconciler", func() {
 			return ready, nil
 		}, timeout*3).Should(BeEquivalentTo(replicas))
 	})
+
+	It("Should set the Paused condition appropriately", func() {
+		replicas := int32(2)
+		labels := map[string]string{"baz": "bar"}
+
+		instance := &machinev1.MachineSet{
+			ObjectMeta: metav1.ObjectMeta{Name: "baz", Namespace: namespace.Name},
+			Spec: machinev1.MachineSetSpec{
+				Replicas: &replicas,
+				Selector: metav1.LabelSelector{
+					MatchLabels: labels,
+				},
+				Template: machinev1.MachineTemplateSpec{
+					ObjectMeta: machinev1.ObjectMeta{
+						Labels: labels,
+					},
+				},
+			},
+		}
+
+		By("Creating the MachineSet")
+		Expect(k8sClient.Create(ctx, instance)).To(Succeed())
+
+		By("Setting the AuthoritativeAPI to ClusterAP")
+		Eventually(k.UpdateStatus(instance, func() {
+			instance.Status.AuthoritativeAPI = machinev1.MachineAuthorityClusterAPI
+		})).Should(Succeed())
+
+		By("Verifying that the AuthoritativeAPI is set to Cluster API")
+		Eventually(func() (machinev1.MachineAuthority, error) {
+			if err := k8sClient.Get(ctx, objectKey(instance), instance); err != nil {
+				return "", err
+			}
+			return instance.Status.AuthoritativeAPI, nil
+		}, timeout).Should(Equal(machinev1.MachineAuthorityClusterAPI))
+
+		By("Verifying the paused condition is approproately set to true")
+		Eventually(func() bool {
+			if err := k8sClient.Get(ctx, objectKey(instance), instance); err != nil {
+				return false
+			}
+
+			// The condition is set to true
+			return conditions.IsTrue(instance, PausedCondition)
+		}, timeout*10).Should(BeTrue())
```

Comment:
If you just returned the conditions objects, this may print better errors when there's an issue
```suggestion
		Eventually(func() ([]machinev1.Condition, error) {
			if err := k8sClient.Get(ctx, objectKey(instance), instance); err != nil {
				return nil, err
			}

			// The condition is set to true
			return instance.Status.Conditions, nil
		}, timeout*10).Should(ContainElement(SatisfyAll(
			HaveField("Type", Equal(PausedCondition)),
			HaveField("Status", BeTrue()),
		)))
```

Also, why `timeout*10`?

Reply:
> Also, why timeout*10?

Ah - I've left my debugging timeouts in. It was so when I was stepping through there would be no failures due to the timeout being exceeded.